src/helpers/ValueHelpers.ts

```diff
@@ -0,0 +1,545 @@
+        return {
+            valueType: "party",
+            party: value
+        };
+    },
+    /**
+     * Wraps a plain boolean into the underlying representation of boolean values on the ledger
+     *
+     * Example:
+     *
+     * ```
+     * daml.boolean(true)
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'bool',
+     *     bool: true
+     * }
+     * ```
+     */
+    bool: (value: boolean): BoolValue => {
+        return {
+            valueType: "bool",
+            bool: value
+        };
+    },
+    /**
+     * If given a {@link Date}, wraps its value into the underlying representation for timestamp.
+     *
+     * If given a number or bigint, casts it to the corresponding string value.
+     *
+     * If given a string, passes through unchanged. Pay extra caution when using this approach.
+     *
+     * _ATTENTION_: plain JavaScript numbers do not suffer from a loss in precision only within the -2^53 to 2^53 range
+     *
+     * Examples:
+     *
+     * ```
+     * daml.timestamp(new Date('1970-01-01T00:00:01Z'))
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'timestamp',
+     *     timestamp: '1000000'
+     * }
+     * ```
+     *
+     * ---
+     *
+     * ```
+     * daml.timestamp('1000000') // strings go through unchanged
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'timestamp',
+     *     timestamp: '1000000'
+     * }
+     * ```
+     *
+     * ---
+     *
+     * ```
+     * daml.timestamp('1970-01-01T00:00:01Z') // CAUTION: STRING GO THROUGH UNCHANGED
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'timestamp',
+     *     timestamp: '1970-01-01T00:00:01Z' // CAUTION: THIS WILL BE A RUNTIME ERROR
+     * }
+     * ```
+     *
+     * ---
+     *
+     * ```
+     * daml.timestamp(1000000)
+     * ```
+     *
+     * becomes
+     *
+     * ```
+     * {
+     *     valueType: 'timestamp',
+     *     timestamp: '1000000'
+     * }
+     * ```
+     */
+    timestamp: (value: Date | string | number | bigint): TimestampValue => {
+        if (value instanceof Date) {
+            const epoch = value.getTime();
+            return {
+                valueType: "timestamp",
+                timestamp: epoch === 0 ? '0' : `${epoch}000`
+            };
+        } else {
+            return {
+                valueType: "timestamp",
+                timestamp: '' + value
+            };
+        }
+    },
+    /**
+     * If given a {@link Date}, wraps its value into the underlying representation for timestamp.
+     *
+     * If given a number or bigint, casts it to the corresponding string value.
+     *
+     * If given a string, passes through unchanged. Pay extra caution when using this approach.
+     *
+     * _ATTENTION_: plain JavaScript numbers do not suffer from a loss in precision only within the -2^53 to 2^53 range
```

Comment:
this is very weird phrasing, i'd much rather say
```suggestion
     * _ATTENTION_: plain JavaScript numbers lose precision outside the [-2^53, 2^53] range.
```